### PR TITLE
add python3-kivy, python3-marshmallow-dataclass-pip and python3-tables to to rosdep/python.yaml

### PR DESCRIPTION
**Pull request by 3473f**
Please add the following dependency to the rosdep database.

## Package name:

python3-kivy
python3-marshmallow-dataclass-pip
python3-tables

## Package Upstream Source:

kivy: https://github.com/kivy/kivy
marshmallow-dataclass: https://github.com/lovasoa/marshmallow_dataclass
tables: https://github.com/PyTables/PyTables

## Purpose of using this:

Kivy: open source Python framework for rapid development of applications that make use of innovative user interfaces, such as multi-touch apps.
Marshmallow-dataclass: Automatic generation of marshmallow schemas from dataclasses.
Tables: managing hierarchical datasets and designed to efficiently and easily cope with extremely large amounts of data. 

Distro packaging links:

## Links to Distribution Packages

Kivy:
- Debian: https://www.debian.org/distrib/packages
  - https://packages.debian.org/bullseye/python3-kivy
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/focal/python3-kivy

marshmallow-dataclass:
- PiPy: https://pypi.org/
  - https://pypi.org/project/marshmallow-dataclass/

PyTables:
- Debian: https://www.debian.org/distrib/packages
  - https://packages.debian.org/bullseye/python3-tables
  - Buster:
    - PiPy: https://pypi.org/
      - https://pypi.org/project/Kivy/
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/python-tables/python3-tables/
- Ubuntu: https://packages.ubuntu.com/
  -  https://packages.ubuntu.com/focal/python3-tables

### CODE REVIEW

File: rosdep/python.yaml [hunk 1]
```diff
@@ -8580,6 +8588,10 @@ python3-systemd:
   debian: [python3-systemd]
   nixos: [python3Packages.systemd]
   ubuntu: [python3-systemd]
+python3-tables-pip:
```
Comment (tfoote) [inline rosdep/python.yaml:8591]: This is packaged natively for example Ubuntu https://packages.ubuntu.com/jammy/python3-tables

Please update this to use the native packages on as many platforms as you can find, as per our contributing guide.

File: rosdep/python.yaml [hunk 2]
```diff
@@ -6955,6 +6955,10 @@ python3-kitchen:
   ubuntu:
     '*': [python3-kitchen]
     xenial: null
+python3-kivy-pip:
```
Comment (tfoote) [inline rosdep/python.yaml:6958]: This is also packaged upstream so pip rules should not be used for it. 

https://packages.ubuntu.com/jammy/python-kivy-examples

**Review by tfoote — CHANGES_REQUESTED** [review]
At least two of these packages have native packages available and as such should be used from those sources not from pip.

The contributing guide lays this out here: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#guidelines-for-rosdep-rules

Please check the various potential platforms for availability and updated your PR to use the native packaging keys. Also don't forget to update the key to remove the `-pip` suffix.

**Comment by 3473f** [comment]
Hello @tfoote, thanks for your feedback. I updated my pull request accordingly.

**Review by tfoote — APPROVED** [review]
Thanks for switching over. It would be great to add more platforms but I didn't find any in a quick search either so lgtm.

**Comment by 3473f** [comment]
> Thanks for switching over. It would be great to add more platforms but I didn't find any in a quick search either so lgtm.

I had to add python3-tables to debian buster through pip as it was not found causing the check to fail.

File: rosdep/python.yaml [hunk 3]
```diff
@@ -6955,6 +6955,13 @@ python3-kitchen:
   ubuntu:
     '*': [python3-kitchen]
     xenial: null
+python3-kivy:
+  debian:
+    '*': [python3-kivy]
+    buster:
+      pip:
+        packages: [Kivy]
```
Comment (jacobperron) [inline rosdep/python.yaml:6963]: I'm not sure we want to mix pip and system packages in the same rosdep rule. We can avoid the warning for buster by making the rule:

```
buster: null
```

Comment (jacobperron) [reply rosdep/python.yaml:6963]: @wjwwood  thoughts?

Comment (3473f) [reply rosdep/python.yaml:6963]: I was following the contribution guide in this case

> Some packages are only available as system packages in newer distributions. In such case, you can specify the default as system package, and pip for older distributions for which the system packages are not available.

Comment (jacobperron) [reply rosdep/python.yaml:6963]: Ah, cool. I found the section: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#pip-mixed-with-system-packages

**Review by wjwwood — APPROVED** [review]

**Pull request merged**